### PR TITLE
NF: Applying Decorators in Module (Segment) 

### PR DESCRIPTION
**Pull request by deka27**
## This PR is to apply keyword decorator to functions in segment module.

This PR is dependent on the main PR, which should be merged first.

## Main PR:
- #3239

### CODE REVIEW

File: dipy/segment/clusteringspeed.pyx [hunk 1]
```diff
@@ -5,6 +5,7 @@ cimport numpy as cnp
 
 from dipy.segment.clustering import ClusterCentroid, ClusterMapCentroid
 from dipy.segment.clustering import TreeCluster, TreeClusterMap
+from dipy.testing.decorators import warning_for_keywords
```
[the remark this answers is not in the document]
Comment (deka27) [reply dipy/segment/clusteringspeed.pyx]: Done.

File: dipy/segment/featurespeed.pyx [hunk 2]
```diff
@@ -5,6 +5,7 @@ cimport numpy as cnp
 
 from dipy.segment.cythonutils cimport tuple2shape, shape2tuple, shape_from_memview
 from dipy.tracking.streamlinespeed cimport c_set_number_of_points, c_length
+from dipy.testing.decorators import warning_for_keywords
```
[the remark this answers is not in the document]
Comment (deka27) [reply dipy/segment/featurespeed.pyx]: Done.

**Comment by deka27** [comment]
This should be good to go

File: dipy/segment/clustering.py [hunk 3]
```diff
@@ -194,6 +197,7 @@ class ClusterMap:
         Actual elements that clustered indices refer to.
     """
 
+    @warning_for_keywords()
     def __init__(self, refdata=None):
```
Comment (skoudoro) [inline dipy/segment/clustering.py:201]: Why `*` is missing here? any reason ?

File: dipy/segment/metricspeed.pyx [hunk 4]
```diff
@@ -7,6 +7,7 @@ from libc.math cimport sqrt, acos
 
 from dipy.segment.cythonutils cimport tuple2shape, shape2tuple, same_shape
 from dipy.segment.featurespeed cimport IdentityFeature
+from dipy.testing.decorators import warning_for_keywords
```
Comment (skoudoro) [inline dipy/segment/metricspeed.pyx:10]: imported but not use so useless. To remove I suppose

**Review by skoudoro — COMMENTED** [review]
Hi @deka27, 

See below the comments. thanks for the future fixing.

**Review by skoudoro — CHANGES_REQUESTED** [review]
Also, reviewing take time, so please, check multiple time before requesting review. it will help both of us

**Review by skoudoro — APPROVED** [review]
All good, thanks @deka27, merging

**Pull request merged**